src/argilla/client/feedback/training/schemas.py

```diff
@@ -1064,13 +1130,208 @@ def _dict_to_format(ds: datasets.Dataset) -> List[Dict[str, List[Dict[str, str]]
+
+    Examples:
+        Example for argilla/emotion dataset:
+        >>> from argilla import TrainingTaskForSentenceSimilarity
+        >>> dataset = rg.FeedbackDataset.from_argilla(name="argilla/emotion")
+        >>> def formatting_func(sample: Dict[str, Any]):
+        ...     return {"sentence": sample["text"], "label": int(sample["label"][0]["value"])}
+        >>> task = TrainingTaskForSentenceSimilarity(formatting_func=formatting_func)
+        >>> dataset.prepare_for_training(framework="...", task=task)
+    """
+
+    _formatting_func_return_types = TrainingTaskForSentenceSimilarityFormat
+    formatting_func: Callable[
+        [Dict[str, Any]],
+        Union[
+            None, Dict[str, Union[float, int]], Dict[str, str], List[Dict[str, Union[float, int]]], List[Dict[str, str]]
+        ],
+    ] = None
+    texts: Optional[List[TextField]] = None
+    label: Optional[LabelQuestionUnification] = None
+
+    @property
+    def supported_frameworks(self):
+        names = ["sentence-transformers"]
+        return [Framework(name) for name in names]
+
+    @property
+    def __all_labels__(self):
+        if self.label:
+            return self.label.question.__all_labels__
+
+    @property
+    def __label2id__(self):
+        if self.label:
+            return self.label.question.__label2id__
+
+    @property
+    def __id2label__(self):
+        if self.label:
+            return self.label.question.__id2label__
+
+    def __repr__(self) -> str:
+        # TODO: Update to include texts/labels when found
+        return f"{self.__class__.__name__}\n\t formatting_func={self.formatting_func}"
+
+    def _format_data(self, dataset: "FeedbackDataset") -> List[Dict[str, Any]]:
+        if self.formatting_func:
+            outputs = []
+            for sample in dataset.format_as("datasets"):
+                output = self.formatting_func(sample)
+                if output is None:
+                    continue
+
+                self._test_output_formatting_func(output)
+
+                outputs.append(output)
+
+            if "label" in outputs[0]:
+                _all_labels = set()
+                for sample in outputs:
+                    if isinstance(sample, list):
+                        for response in sample:
+                            _all_labels.add(response["label"])
+                    else:
+                        _all_labels.add(sample["label"])
+
+                self.label = LabelQuestionUnification(
+                    question=LabelQuestion(name="custom_func", labels=list(_all_labels))
+                )
+
+            return outputs
+
+        else:
+            formatted_data = super()._format_data(dataset)
+            # NOTE: Maybe this post processing of the formatted data can be simplified
+            # or directly done in super()._format_data(dataset).
+            new_keys = {field.name: f"sentence-{i+1}" for i, field in enumerate(self.texts)}
```

Comment:
```suggestion
            new_keys = {field.name: f"sentence-{i}" for i, field in enumerate(self.texts, start=1)}
```
Extremely nitpicky, but I usually prefer this solution 😄